### PR TITLE
first version to request profile merges

### PR DESCRIPTION
**Pull request by melisabok**
- Use invitation openreview.net/Support/-/Profile_Merge to request a profile merge. 
- Logged users and guest users can make this request.
- They need to specify the left and right email/tilde id.
- When moderating the requests, the UI should show a link to the profile/compare with the left and right parameters.
- After the moderator merge the profiles, the moderator needs to click on "Accept" to complete the request and notify the user.
- The moderator can also rejects the request if one of the left/right is wrong or there is no evidence the user is the owner of the profiles. 

UPDATE:

- set dblp invitations
- fix author and authors matching the same name in the author name deletion process.

### CODE REVIEW

File: openreview/profile/management.py
Comment (carlosmondra) [inline openreview/profile/management.py:236]:
```suggestion
                'description': 'Username or email that want to be merged.',
```

File: openreview/profile/management.py
Comment (carlosmondra) [inline openreview/profile/management.py:242]:
```suggestion
                'description': 'Username or email that want to be merged.',
```

File: openreview/profile/management.py
Comment (carlosmondra) [inline openreview/profile/management.py:233]: Why are we including this field in this invitation?

File: openreview/profile/process/request_merge_profiles_decision_process.py
Comment (carlosmondra) [inline openreview/profile/process/request_merge_profiles_decision_process.py:19]: We need to change this line, right?

> We can not remove the name from the profile for the following reason:

File: openreview/profile/management.py
Comment (melisabok) [reply openreview/profile/management.py:233]: sometimes guest users are requesting merge of profiles, and we would need to have a name.

File: openreview/profile/process/request_merge_profiles_decision_process.py
Comment (melisabok) [reply openreview/profile/process/request_merge_profiles_decision_process.py]: fixed

**Comment by carlosmondra** [comment]
> > Where are guest users going to do the request from ?
> 
> Some users from the signup page? i'm not sure who to do here but we receive emails with the merge profile request from guest users. Sometimes they lost the access to the profile?

What if we ask users to first register by either claiming a profile or creating a new one and then request a merge?

**Comment by xkopenreview** [comment]
> > > Where are guest users going to do the request from ?
> > 
> > 
> > Some users from the signup page? i'm not sure who to do here but we receive emails with the merge profile request from guest users. Sometimes they lost the access to the profile?
> 
> What if we ask users to first register by either claiming a profile or creating a new one and then request a merge?

i think there are 3 places where user can request a merge:
1. sign up page there's a link at bottom to send feedback
2. when user add a email which belong to another profile
3. when user import dblp paper associated with another profile

i think we can check the title of the message to know whether it's profile merge or not

**Comment by carlosmondra** [comment]
> i think there are 3 places where user can request a merge:
> 
> 1. sign up page there's a link at bottom to send feedback
> 2. when user add a email which belong to another profile
> 3. when user import dblp paper associated with another profile
> 
> i think we can check the title of the message to know whether it's profile merge or not

Right, I was hoping to remove option 1, so that only registered users can request merges.

**Comment by melisabok** [comment]
Should guest users contact us if they want to merge profiles or recover a profile where they lost access to the email account?

**Comment by xkopenreview** [comment]
> > i think there are 3 places where user can request a merge:
> > 
> > 1. sign up page there's a link at bottom to send feedback
> > 2. when user add a email which belong to another profile
> > 3. when user import dblp paper associated with another profile
> > 
> > i think we can check the title of the message to know whether it's profile merge or not
> 
> Right, I was hoping to remove option 1, so that only registered users can request merges.

i think the user action following or before 1 is to claim a profile/resend activation email/reset password
so it would save a tilde id to allow guest to request profile merge.

what's the rational to only allow logged in user to request merge?

**Comment by carlosmondra** [comment]
> Should guest users contact us if they want to merge profiles or recover a profile where they lost access to the email account?

I think users should register with whatever profile they can claim and then request a merge.
Recovering profiles is a separate matter right? Or what am I missing?

> i think the user action following or before 1 is to claim a profile/resend activation email/reset password
> so it would save a tilde id to allow guest to request profile merge.

I think it would be better for users to claim a profile/resend activation email/reset password and then, after the profile is active, the user can request profile merges.

> what's the rational to only allow logged in user to request merge?

To have profile merge requests in a single place.

I'm not sure where the request to merge profiles is going to be created, but if, for example, we are going to use the profile page for that, then, guest users won't have a profile page to do it and I want to avoid that.

**Comment by xkopenreview** [comment]
currently 1 and 2 are using the feedback modal 
i thought we can just change it to use a new profile merge request modal
but just realized that 3 just instruct user to contact us and it's already a modal and it's not good to have modal open new modal so maybe 3 will be in profile edit page

File: openreview/profile/management.py
Comment (xkopenreview) [reply openreview/profile/management.py:233]: maybe this should be email instead of name
so that the user can be contacted if we need clarification

File: tests/test_profile_management.py
Comment (xkopenreview) [inline tests/test_profile_management.py:1061]: i think instead of just accept or reject, the direction of merge also need to be specified

Comment (carlosmondra) [reply tests/test_profile_management.py:1061]: Maybe the direction should be for us to decide

Comment (xkopenreview) [reply tests/test_profile_management.py:1061]: discussed this in the daily meeting.
the actual profile merge will be done by the moderator in profile compare page(same as how it's done currently)

the user moderation page will have a link to profile compare page and after the profile is merged the moderator will go back to moderation page to click approve to indicate the profile merge has been done.(or don't merge and reject)

File: openreview/profile/management.py
Comment (xkopenreview) [reply openreview/profile/management.py:233]: we discussed about combining signature and email column
so this field need to be renamed to email and required set to false

File: openreview/profile/management.py
Comment (xkopenreview) [inline openreview/profile/management.py:304]: maybe this should be a regex for email

File: openreview/profile/management.py
Comment (celestemartinez) [inline openreview/profile/management.py:322]: Should we change this to: `Reason why you want to merge your profiles.` (though I'm sure everyone wants to merge because they have multiple profiles). 
Maybe just `Any additional information about the merge.`

Comment (melisabok) [reply openreview/profile/management.py:322]: good idea, @xkopenreview is the UI using this description?

Comment (xkopenreview) [reply openreview/profile/management.py:322]: no currently the label of input is fixed.

File: openreview/profile/management.py
Comment (melisabok) [reply openreview/profile/management.py]: fixed!

**Review by xkopenreview — APPROVED** [review]

**Review by carlosmondra — APPROVED** [review]

**Review by celestemartinez — APPROVED** [review]

**Pull request merged**
